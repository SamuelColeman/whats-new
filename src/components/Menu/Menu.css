button {
	height: 5%;
  border: none;
  margin: 2%;
  background: #141527;
  color: white;
  font-size: 20px;
}

button:hover {
	background: #FFAD03;
	border-radius: 0 15px 15px 0;
	box-shadow: 5px 5px 5px black;
}

button:focus {
	outline: 0;
}