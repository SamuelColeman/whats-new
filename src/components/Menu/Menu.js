import React, { Component } from 'react';
import './Menu.css'

const Menu = props => {
	return (
		<>
			<button onClick={() => props.filterArticles('local')} type="button">Local News</button>
	    <button onClick={() => props.filterArticles('technology')} type="button">Technology</button>
	    <button onClick={() => props.filterArticles('entertainment')} type="button">Entertainment</button>
	    <button onClick={() => props.filterArticles('science')} type="button">Science</button>
	    <button onClick={() => props.filterArticles('health')} type="button">Health</button>
	  </>
	)
}


export default Menu;