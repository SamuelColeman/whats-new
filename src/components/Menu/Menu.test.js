import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
	const mockFilterArticles = jest.fn();
	const wrapper = shallow(<Menu filterArticles={mockFilterArticles}/>);

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	})

	it('should invoke filterArticles when a button is clicked', () => {
			wrapper.find('button').at(0).simulate('click');
			expect(mockFilterArticles).toHaveBeenCalledWith('local');
	})
})