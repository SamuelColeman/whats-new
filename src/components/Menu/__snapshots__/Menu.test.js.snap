// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
<Fragment>
  <button
    onClick={[Function]}
    type="button"
  >
    Local News
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Technology
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Entertainment
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Science
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Health
  </button>
</Fragment>
`;
